WordPress/gutenberg
# Site editor: Fix save shortcut

## What?
While working on some shortcuts issue I noticed that we in site editor we register the `Save your changes.` twice. There is a reason for needing a special save shortcut in site editor which is to handle saving when we are in `view` mode.

This PR fixes two issues:
1. In view mode when we have changes only in the current edited entity, the shortcut now works and saves the entity.
2. The special site editor `Save your changes` shortcut is only registered in view mode and therefore is not duplicate in the respective list in the dialog.


## Screenshots

Check the `Save your changes.` command(s).

Before            |  After
:-------------------------:|:-------------------------:
<img width="878" alt="before" src="https://github.com/user-attachments/assets/9252c4f9-b96e-448d-9831-a84a9d2febca"> | <img width="878" alt="after" src="https://github.com/user-attachments/assets/ed411723-3b58-4300-b78c-3cee00df5b48">


 



Previously, I believe issue `number 1` used to work and probably the code was somehow reorganised. Not that it matters much..



## Testing Scenarios in site editor

1. Make some changes to an entity and go to `view` mode without saving. The shortcut for save should work properly.
2. Make some changes to an entity and go to `view` mode without saving. Make changes to a second entity and go back to view mode. The shortcut should open the `dialog` to save.
4. Multiple entity changes should work as before and open the dialog in view mode.
5. In all other cases the shortcut should work as before.
